Cleanup files

islipfd19 opened the pull request:
Cleanup files and prepare for PR

awiouy committed:
libestr: initial package

awiouy committed:
libfastjson: initial package

awiouy committed:
liblogging: initial package

awiouy committed:
liblognorm: initial package

awiouy committed:
librelp: initial package

awiouy committed:
rsyslog: initial addon

islipfd19 committed:
update files

awiouy commented on packages/addons/service/rsyslog/source/settings-default.xml:
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8" standalone="yes"?> <settings>
+    <category label="1000">
```
No category in saved settings

islipfd19 replied:
Remove line #2 all together...? Not sure how '1000' stayed in there. Thought I had updated that.

awiouy commented on packages/addons/service/rsyslog/source/settings-default.xml:
```diff
@@ -0,0 +1,11 @@
+    <category label="1000">
+		<setting label="30010" type="lsep"/>
+		<setting type="sep" />
+		<setting id="RSYSLOG_IP_NAME_FQDN" type="text" label="30011" values="" default="" />
```
label, type, id would be cleaner
text, number and bool have no values attribute
http://kodi.wiki/view/Add-on_settings

islipfd19 replied:
Remove all 'values=""' verbiage from all text, number and bool lines, right?

awiouy replied:
eg https://github.com/LibreELEC/LibreELEC.tv/blob/master/packages/addons/service/spotify-connect-web/source/resources/settings.xml

awiouy commented on packages/addons/service/rsyslog/source/resources/settings.xml:
```diff
@@ -1,9 +1,12 @@
-      <setting label="30003" type="lsep" />
-      <setting label="30004" type="lsep" />
+   <category label="30010">
+      <setting label="30011" type="lsep"/>
```
Move this to category

islipfd19 replied:
Your speaking to 'type="lsep"/>'? That should be moved to category?

awiouy replied:
RSYSLOG is already displayed by the UI, no need to repeat

awiouy commented on packages/addons/service/rsyslog/source/bin/rsyslog.init:
```diff
@@ -20,9 +20,95 @@
-if [ ! -f "$ADDON_HOME/rsyslog.conf" ]
+ADDON_SETTINGS="$ADDON_HOME/settings.xml"
+ADDON_DEST="$ADDON_HOME/conf"
+RSYSLOG_IP_NAME_FQDN=`grep RSYSLOG_IP_NAME_FQDN $ADDON_SETTINGS | awk '{print $3 }' | sed -e "s,value=,," -e "s,\",,g"`
```
oe_setup_addon takes care of this and the following lines
https://github.com/LibreELEC/LibreELEC.tv/blob/ce890b1f04c0932f3d9912eaaf5fab237f6fd38f/packages/mediacenter/kodi/profile.d/00-addons.conf

islipfd19 replied:
So all lines beginning with 'RSYSLOG_' should be removed?

awiouy replied:
Look at the code ;)

awiouy commented on packages/addons/service/rsyslog/source/bin/rsyslog.init:
```diff
@@ -20,9 +20,95 @@
+fi
+
+cat > "$ADDON_HOME/rsyslog.conf" << 'RSYSLOG_CONF_BLOCK_PRE'
+# rsyslog v5 configuration file
```
Maybe add $ModDir (?) so that rsyslog finds modules even from command line?

islipfd19 replied:
I know what you mean but not sure how it should be done?

awiouy replied:
Not sure if this is still supported
http://www.rsyslog.com/doc/v5-stable/configuration/global/options/rsconf1_moddir.html

islipfd19 committed:
Cleanup and format according to Kodi spec

Removed unwarranted 'values=""' from strings and reorganized structure.

islipfd19 committed:
Cleanup and format according to Kodi spec

Removed unwarranted 'values=""' from strings and reorganized structure.

islipfd19 committed:
Update msgctxt values

Updated msgctxt values to match that of the associated settings.xml and settings-default.xml files to match corresponding entries

islipfd19 committed:
Clean and don't be redundant

00-addons.conf parses appropriate addon settings key values. No need to duplicate what it already does.

islipfd19 committed:
Fix selector for UDP/TCP

islipfd19 commented:
Built and tested on WeTekHub. Working as expected

awiouy commented:
Implemented differently. Closing
